src/service/models.py

```diff
@@ -102,3 +102,32 @@ class HealthResponse(BaseModel):
     """Health check response model."""
 
     status: Annotated[str, Field(description="Health status")]
+
+
+class DeploymentStatus(BaseModel):
+    """Status information of a Kubernetes deployment."""
+
+    available_replicas: Annotated[
+        int, Field(description="Number of available replicas")
+    ] = 0
+    ready_replicas: Annotated[int, Field(description="Number of ready replicas")] = 0
```

Comment:
What's the difference between a ready replica and an available replica?

Reply:
my understanding is in order for a pod to be in the available state, it first needs to be ready and it should stay as ready for a period of time (defined as `minReadySeconds`).

Reply:
Ok. I'm wondering if it's worth exposing the distinction to users, I'd assume all they are about is the ready state

Reply:
My current design is to have a wrapper method in Jupyterhub to call APIs in this repo. We can probably filter some status while users calling the wrapper method. At the same time,  developers can call API directly and have access to see more informative status for troubleshooting purposes.

Reply:
:+1: